File: dotnet/Lexy.Poc/Lexy.Poc.Core/Compiler/Transcribe/Modifiers.cs
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.CSharp;

namespace Lexy.Poc.Core.Transcribe
{
    internal class Modifiers
    {
        public static SyntaxToken Public => SyntaxFactory.Token(SyntaxKind.PublicKeyword);
        public static SyntaxToken Static => SyntaxFactory.Token(SyntaxKind.StaticKeyword);
        public static SyntaxTokenList PublicAsList => SyntaxTokenList.Create(Public);
        public static SyntaxTokenList StaticAsList => SyntaxTokenList.Create(Static);
    }
}

File: typescipt/src/compiler/cSharp/modifiers.ts


internal static class Modifiers {
   private static publicToken(): SyntaxToken {
     return Token(SyntaxKind.PublicKeyword);
   }

   private static privateToken(): SyntaxToken {
     return Token(SyntaxKind.PrivateKeyword);
   }

   private static staticToken(): SyntaxToken {
     return Token(SyntaxKind.StaticKeyword);
   }

   public static public(): SyntaxTokenList {
     return TokenList(PublicToken());
   }

   public static static(): SyntaxTokenList {
     return TokenList(StaticToken());
   }

   public static private(): SyntaxTokenList {
     return TokenList(PrivateToken());
   }

   public static privateStatic(): SyntaxTokenList {
     return TokenList(PrivateToken(), StaticToken());
   }

   public static publicStatic(): SyntaxTokenList {
     return TokenList(PublicToken(), StaticToken());
   }
}
